src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -67,6 +67,7 @@ import { useDiscoverCustomization } from '../../../../customizations';
 import { onResizeGridColumn } from '../../../../utils/on_resize_grid_column';
 import { useContextualGridCustomisations } from '../../hooks/grid_customisations';
 import { useIsEsqlMode } from '../../hooks/use_is_esql_mode';
+import { useAdditionalFieldGroups } from '../../hooks/sidebar/use_additional_field_groups';
```

Comment:
Should the returned content of `useAdditionalFieldGroups` be memoized? Currently it will return a new reference to to the {smartFields, fallbackFields} every time this hook is called, causing all the components with `additionalFieldGroups` to re-render again even though content of `smartFields` and `fallbackFields` are the same.

Reply:
Yeah, I'll add some memoization. That hook is guaranteed to change soon based on "contextual awareness" landing in Discover, but I can certainly add something for now 👍